streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -237,7 +237,7 @@ private static boolean isRecoverable(final KafkaException uncaughtException) {
      * @throws IllegalStateException if EOS is disabled
      * @throws TaskMigratedException
      */
-    void commitTransaction(final Map<TopicPartition, OffsetAndMetadata> offsets,
+    protected void commitTransaction(final Map<TopicPartition, OffsetAndMetadata> offsets,
```

Comment:
Needed this so I could move the subclass to a different package. I think it's still just as obviously inappropriate for users to subclass this class, since it's in the `internals` package.